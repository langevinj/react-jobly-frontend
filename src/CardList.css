.CardList{
    margin-top: 100px;
    margin-right: 25%;
    margin-left: 25%;
    text-align: center;
}