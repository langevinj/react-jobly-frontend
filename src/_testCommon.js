const jobPages = [[{
    companyHandle: "mejia-scott-ryan",
    companyName: "Mejia, Scott and Ryan",
    equity: null,
    id: 200,
    salary: 126000,
    title: "Accommodation manager"
}, {
    companyHandle: "stone-stewart",
    companyName: "Stone-Stewart",
    equity: "0",
    id: 36,
    salary: 175000,
        title: "Accountant, chartered certified"
    }, {
        companyHandle: "boyd-evans",
        companyName: "Boyd-Evans",
        equity: "0.070",
        id: 161,
        salary: 86000,
        title: "Accountant, chartered certified"}]];



const u1 = {user: {username: "testuser", firstName: "test", lastName: "testy", isAdmin: false, applications: [jobPages[0][0].id, jobPages[0][1].id]}}

// const companies = { companies: [{ handle: "test-company-1", name="testcompany1", description: "the first test company.", numEmployees: 50, logoUrl: "www.image.jpg" }, { handle: "test-company-2", name="testcompany2", description: "the second test company.", numEmployees: 100, logoUrl: "www.fake.com" }] }

export { jobPages, u1 }