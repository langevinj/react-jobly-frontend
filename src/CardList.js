import React from 'react'
import Card from './Card'
import './CardList.css'

function CardList({ title, items = [], apply = () => null, unapply = () => null, pageNum = 0}) {
    //if absolutely no results are found, tell the user, otherwise list as many items as the page provides
    const pageItems = items[0] && Array.isArray(items[0]) ? items[pageNum] : items 
    return items.length ? (
        <div className="CardList">
            {pageItems.map((item, idx) => item ? (<Card key={idx} id={item.id} title={title} item={item} apply={apply} unapply={unapply}/>) : null)}
        </div>
    ) : (
        <p>Sorry, no results found!</p>
    )
}

export default CardList;