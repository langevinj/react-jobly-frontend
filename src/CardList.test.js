import React from 'react'
import { render } from '@testing-library/react';
import '@testing-library/jest-dom'
import { MemoryRouter } from 'react-router-dom'
import { jobPages } from './_testCommon'
import CardList from './CardList'

it("renders without crashing", function() {
    render(<CardList />);
});

it("matches the snapshot", function() {
    const { asFragment } = render(<CardList items={["testcard1", "testcard2"]}/>)

    expect(asFragment()).toMatchSnapshot();
});