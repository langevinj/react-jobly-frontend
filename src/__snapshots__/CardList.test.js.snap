// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="CardList"
  >
    <div
      class="Card-body"
    >
      <div
        class="JobCard"
      >
        <h6 />
        <p>
          Salary: 
        </p>
        <p>
          Equity: 
        </p>
        <button
          class="Apply btn-primary rounded"
          data-toggle="button"
          type="button"
        >
          Apply
        </button>
      </div>
    </div>
    <div
      class="Card-body"
    >
      <div
        class="JobCard"
      >
        <h6 />
        <p>
          Salary: 
        </p>
        <p>
          Equity: 
        </p>
        <button
          class="Apply btn-primary rounded"
          data-toggle="button"
          type="button"
        >
          Apply
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
